### PR TITLE
Update GSON library to 2.9.0

### LINKED ISSUE #21126: Update GSON library dependency to 2.9.0
Updates the usage of GSON to version 2.9.0.

### PR DESCRIPTION
**Pull request by ayoho**
Resolves #21126

### CODE REVIEW

**Comment by ayoho** [comment]
#build

**Comment by ayoho** [comment]
#build

**Comment by ayoho** [comment]
#libby
#build

**Comment by ayoho** [comment]
#libby
#build

**Comment by ayoho** [comment]
#build

**Comment by ayoho** [comment]
#libby
#build

**Comment by ayoho** [comment]
#libby

**Comment by ayoho** [comment]
#build

**Comment by ayoho** [comment]
#libby

**Comment by ayoho** [comment]
#build

File: dev/cnf/oss_source_dependencies.maven [hunk 1]
```diff
@@ -4,7 +4,6 @@ com.fasterxml:classmate:1.3.4
 com.google.api.grpc:proto-google-common-protos:2.0.1
 com.google.code.findbugs:jsr305:3.0.2
 com.google.code.gson:gson:2.2.4
```
Comment (tevans78) [inline dev/cnf/oss_source_dependencies.maven:6]: Don't you want to also remove 2.2.4 from this file and only use the 2.9.0 version?

File: dev/com.ibm.ws.cdi.third.party_fat/build.gradle [hunk 2]
```diff
@@ -32,7 +32,7 @@ dependencies {
     project(':com.ibm.ws.org.slf4j.jdk14'),
     'net.sf.jtidy:jtidy:r938'
   hibernateSearchLibs 'org.antlr:antlr:3.5.2',
-    'com.google.code.gson:gson:2.8.5',
+    'com.google.code.gson:gson:2.9.0',
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.third.party_fat/build.gradle]: shouldn't this reference be changed to `project('com.ibm.ws.com.google.gson')`?

File: dev/cnf/oss_source_dependencies.maven (hunk 1 above)
Comment (tevans78) [reply dev/cnf/oss_source_dependencies.maven:6]: as per slack, assume you'll delete this line later

File: dev/com.ibm.ws.cdi.third.party_fat/build.gradle (hunk 2 above)
Comment (tevans78) [reply dev/com.ibm.ws.cdi.third.party_fat/build.gradle]: or even better... `project('io.openliberty.com.google.gson')`

**Comment by ayoho** [comment]
#build

**Comment by ayoho** [comment]
#libby
#build

File: dev/com.ibm.ws.microprofile.opentracing.jaeger_fat/bnd.bnd [hunk 3]
```diff
@@ -43,7 +43,7 @@ tested.features: jaeger-1.0
     com.ibm.ws.javaee.servlet.3.1,\
     javax.servlet-api,\
     org.apache.geronimo.specs.geronimo-servlet_3.0_spec,\
-    com.google.code.gson:gson;version=2.8.2,\
+    com.google.code.gson:gson;version=2.9.0,\
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.opentracing.jaeger_fat/bnd.bnd]: Is there a reason not to do it this way? The more places that point at the bundle, the less that need updating next time...
```suggestion
    io.openliberty.com.google.gson;version=latest,\
```

File: dev/com.ibm.ws.security.oauth_test.custom_servlets/test-applications/oAuth20MongoSetupEar/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -1,5 +1,5 @@
 Manifest-Version: 1.0
 Ant-Version: Apache Ant 1.7.1
-Class-Path: gson-2.2.4.jar guava-30.1-android
+Class-Path: gson-2.9.0.jar guava-30.1-android
```
Comment (tevans78) [inline dev/com.ibm.ws.security.oauth_test.custom_servlets/test-applications/oAuth20MongoSetupEar/META-INF/MANIFEST.MF]: are you sure that this Class-Path is right? Where do these jars come from?
The gradle for this project seems to pull in a jar from elsewhere but not under this name
https://github.com/OpenLiberty/open-liberty/blob/9b0205e67f541078c5e2243da85f524ee3805a7e/dev/com.ibm.ws.security.oauth_test.custom_servlets/build.gradle#L70

File: dev/com.ibm.ws.session.cache_fat_config_infinispan/build.gradle [hunk 5]
```diff
@@ -26,7 +26,7 @@ configurations {
 // Define G:A:V coordinates of each dependency
 dependencies {
   caffeine 'com.github.ben-manes.caffeine:caffeine:2.8.0'
-  gson 'com.google.code.gson:gson:2.8.2'
+  gson 'com.google.code.gson:gson:2.9.0'
```
Comment (tevans78) [inline dev/com.ibm.ws.session.cache_fat_config_infinispan/build.gradle]:
```suggestion
  gson project(':io.openliberty.com.google.gson')
```

File: dev/com.ibm.ws.session.cache_fat_infinispan/build.gradle [hunk 6]
```diff
@@ -26,7 +26,7 @@ configurations {
 // Define G:A:V coordinates of each dependency
 dependencies {
   caffeine 'com.github.ben-manes.caffeine:caffeine:2.8.0'
-  gson 'com.google.code.gson:gson:2.8.2'
+  gson 'com.google.code.gson:gson:2.9.0'
```
Comment (tevans78) [inline dev/com.ibm.ws.session.cache_fat_infinispan/build.gradle]:
```suggestion
  gson project(':io.openliberty.com.google.gson')
```

File: dev/io.openliberty.com.google.gson/bnd.overrides [hunk 7]
```diff
@@ -0,0 +1,14 @@
+-include= ~../cnf/resources/bnd/rejar.props
+bVersion=1.0
+
+Bundle-Name: google.collection
+Bundle-Description: google.collection; version=2.9.0
+Bundle-SymbolicName: io.openliberty.com.google.gson
+
+WS-TraceGroup: OPENIDCONNECT
```
Comment (tevans78) [inline dev/io.openliberty.com.google.gson/bnd.overrides]: do you really want this trace group here?

File: dev/io.openliberty.microprofile.opentracing.2.0.internal.jaeger_fat/bnd.bnd [hunk 8]
```diff
@@ -37,7 +37,7 @@ tested.features: mpOpenTracing-2.0
     com.ibm.websphere.javaee.interceptor.1.2;version=latest,\
     com.ibm.ws.cdi.interfaces;version=latest,\
     com.ibm.ws.componenttest,\
-    com.google.code.gson:gson;version=2.8.2,\
+    com.google.code.gson:gson;version=2.9.0,\
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.opentracing.2.0.internal.jaeger_fat/bnd.bnd]:
```suggestion
    io.openliberty.com.google.gson;version=latest,\
```

**Comment by ayoho** [comment]
#build

**Comment by ayoho** [comment]
#libby

**Comment by ayoho** [comment]
#libby
#build

**Comment by ayoho** [comment]
#build

**Comment by ayoho** [comment]
#libby
#build

File: dev/cnf/oss_source_dependencies.maven [hunk 9]
```diff
@@ -4,7 +4,7 @@ com.fasterxml:classmate:1.3.4
...
-com.google.code.gson:gson:2.8.2
+com.google.code.gson:gson:2.9.0
```
Comment (tevans78) [inline dev/cnf/oss_source_dependencies.maven:7]: Are there still any references to version 2.2.4?

**Review by tevans78 — APPROVED** [review]
Approved but it would still be a good idea to remove references to version 2.2.4 in a follow up PR ... although that might require changes in CL as well.

**Pull request merged**